src/components/utils/caseUtils.ts

```diff
@@ -31,48 +32,48 @@ const downloadCases = async (uuids: string[]) => {
     }
 };
 
-const exportCases = async (
-    cases: any[],
-    format: string,
-    formatParameters: {
-        [parameterName: string]: any;
-    },
-    onError?: (caseElement: any, errorMsg: string) => void
-): Promise<void> => {
-    const files: { name: string; blob: Blob }[] = [];
-    for (const c of cases) {
+export function useDownloadUtils() {
+    const intl = useIntl();
+    const { snackError, snackInfo } = useSnackMessage();
+    const capitalizeFirstLetter = (string: string) =>
+        `${string.charAt(0).toUpperCase()}${string.slice(1)}`;
+    const [abortController, setAbortController] = useState<AbortController>();
+
+    const exportCase = async (
+        caseElement: any,
+        format: string,
+        formatParameters: {
+            [parameterName: string]: any;
+        },
+        abortController: AbortController
+    ): Promise<void> => {
         try {
-            const result = await exportCase(
-                c.elementUuid,
+            const result = await fetchConvertedCase(
+                caseElement.elementUuid,
                 format,
-                formatParameters
+                formatParameters,
+                abortController
             );
             let filename = result.headers
                 .get('Content-Disposition')
                 .split('filename=')[1];
             filename = filename.substring(1, filename.length - 1); // We remove quotes
             const blob = await result.blob();
-            files.push({ name: filename, blob });
-        } catch (e: any) {
-            onError?.(c, e);
-        }
-    }
-    for (const file of files) {
-        const href = window.URL.createObjectURL(file.blob);
-        const link = document.createElement('a');
-        link.href = href;
-        link.setAttribute('download', file.name);
-        document.body.appendChild(link);
-        link.click();
-        document.body.removeChild(link);
-    }
-};
 
-export function useDownloadUtils() {
-    const intl = useIntl();
-    const { snackError, snackInfo } = useSnackMessage();
-    const capitalizeFirstLetter = (string: string) =>
-        `${string.charAt(0).toUpperCase()}${string.slice(1)}`;
+            const href = window.URL.createObjectURL(blob);
+            const link = document.createElement('a');
+            link.href = href;
+            link.setAttribute('download', filename);
+            document.body.appendChild(link);
+            link.click();
+            document.body.removeChild(link);
+        } catch (error: any) {
+            if (error.name === 'AbortError') {
+                throw error;
+            }
+            handleCaseExportError?.(caseElement, error);
```

Comment:
why do use use '?' here ? handleCaseExportError seem to be always defined

Reply:
Indeed. Done.